website/source/docs/vault-integration/index.html.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: "docs"
+page_title: "Vault Integration"
+sidebar_current: "docs-vault-integration"
+description: |-
+  Learn how to integrate with HashiCorp Vault and retrieve Vault tokens for
+  tasks.
+---
+
+# Vault Integration
+
+Many workloads require access to tokens, passwords, certificates, API keys, and
+other secrets. To enable secure, auditable and easy access to your secrets,
+Nomad integrates with HashiCorp's [Vault][]. Nomad Servers and Clients
+coordinate with Vault to derive a Vault token that has access to only the Vault
+policies the tasks needs. Nomad Clients make the token avaliable to the task and
+handle the tokens renewal. Further, Nomad's [`template` block][template] can
+retrieve secrets from Vault making it easier than ever to secure your
+infrastructure. 
+
+Note that in order to use Vault with Nomad, you will need to configure and
+install Vault separately from Nomad. Nomad does not run Vault for you.
+
+## Vault Configuration
+
+In order to use the Vault integration, Nomad Servers must be given a Vault
+token. This Vault token can be either a root token or a token created from a
+role. The root token provides an easy way to get started but it is recommended
+to use the role based token described below. If the token is periodic, Nomad
+Servers will renew the token.
+
+### Root Token
+
+If Nomad is given a root token, no further configuration is needed as Nomad can
+derive a token for jobs using any Vault policies.
+
+### Role based Token
+
+Vault's [Token Authentication Backend][auth] supports a concept called "roles".
+Roles allow policies to be grouped together and token creation to be delegated
+to a trusted service such as Nomad. By creating a role, the set of policies that
+task's managed by Nomad can acess may be limited compared to giving Nomad a root
+token.
+
+When given a non-root token, Nomad queries the token to determine the role it
+was generated from. It will then derive tokens for jobs based on that role.
+Nomad expects the role to be created with several properties described below
+when creating the role with the Vault endpoint `/auth/token/roles/<role_name>`:
+
+```json
+{
+    "allowed_policies": "<comma-seperated list of policies>",
+    "explicit_max_ttl": 0,
+    "name": "nomad",
+    "orphan": false,
+    "period": 259200,
+    "renewable": true
+}
+```
+
+#### Parameters: 
+
+* `allowed_policies`: The `allowed_policies` is a comma separated list of
```

Comment:
- Can you follow the existing pattern of listing the types here
- Use `-` instead of `:` between the thing and the description
- Start the description with "Specifies" across all these values

Thanks!